processor/k8sprocessor/processor.go

```diff
@@ -33,43 +36,72 @@ const (
 )
 
 type kubernetesprocessor struct {
-	logger          *zap.Logger
-	apiConfig       k8sconfig.APIConfig
-	nextConsumer    consumer.TraceConsumer
-	kc              kube.Client
-	passthroughMode bool
-	rules           kube.ExtractionRules
-	filters         kube.Filters
+	logger              *zap.Logger
+	apiConfig           k8sconfig.APIConfig
+	kc                  kube.Client
+	passthroughMode     bool
+	rules               kube.ExtractionRules
+	filters             kube.Filters
+	nextTraceConsumer   consumer.TraceConsumer
+	nextMetricsConsumer consumer.MetricsConsumer
 }
 
 var _ (component.TraceProcessor) = (*kubernetesprocessor)(nil)
+var _ (component.MetricsProcessor) = (*kubernetesprocessor)(nil)
 
-// NewTraceProcessor returns a component.TraceProcessorOld that adds the WithAttributeMap(attributes) to all spans
+// NewTraceProcessor returns a component.TraceProcessor that adds the WithAttributeMap(attributes) to all spans
 // passed to it.
 func NewTraceProcessor(
 	logger *zap.Logger,
-	nextConsumer consumer.TraceConsumer,
+	nextTraceConsumer consumer.TraceConsumer,
 	kubeClient kube.ClientProvider,
 	options ...Option,
 ) (component.TraceProcessor, error) {
-	kp := &kubernetesprocessor{logger: logger, nextConsumer: nextConsumer}
+	kp := &kubernetesprocessor{logger: logger, nextTraceConsumer: nextTraceConsumer}
 	for _, opt := range options {
 		if err := opt(kp); err != nil {
 			return nil, err
 		}
 	}
+	err := kp.initKubeClient(logger, kubeClient)
```

Comment:
nit: I feel it is slightly better if the function creates a client instance and returns it leaving it to the called to set it to a field. Not a big deal though in this case.

Reply:
Agreed. The PR merged before I pushed change for this. Created another PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/361

Reply:
I tried this way, but it seems adding more unnecessary complexity making this code reuse case not very useful.